### PR TITLE
config-loader(schema/filtering): uses key visibility as fallback

### PR DESCRIPTION
**Pull request by dnsaoki2**
Hi,

This modification is due to the PR #8237.

### CODE REVIEW

File: packages/config-loader/src/lib/schema/filtering.ts [hunk 1]
```diff
@@ -61,11 +61,25 @@ export function filterByVisibility(
...
       for (const [index, value] of jsonVal.entries()) {
-        const out = transform(
-          value,
+        const hasVisibilityInIndex = visibilityByDataPath.get(
           `${visibilityPath}/${index}`,
-          `${filterPath}[${index}]`,
         );
+        let out;
+
+        if (typeof value !== 'object' && !hasVisibilityInIndex) {
```
Comment (Rugvip) [inline packages/config-loader/src/lib/schema/filtering.ts:69]: Wonder if we could tweak this logic a little to make it slightly easier to follow? If we figure out the visibilityPath first we can get away with just a single call to `transform`.

**Review by Rugvip — COMMENTED** [review]
Thank you, looks good! 👍 

Thinking we can tweak it a lil bit? but pretty much good to go

File: packages/config-loader/src/lib/schema/filtering.ts (hunk 1 above)
Comment (dnsaoki2) [reply packages/config-loader/src/lib/schema/filtering.ts]: @Rugvip yes of course. I made a change, what do you think?

**Review by Rugvip — APPROVED** [review]
Yep, looks a bit simpler, thanks! 👍

**Comment by dnsaoki2** [comment]
@Rugvip thanks :smile:

**Pull request merged**